analyzers/src/SonarAnalyzer.CSharp/Rules/MethodsShouldNotHaveIdenticalImplementations.cs

```diff
@@ -58,11 +58,12 @@ public sealed class MethodsShouldNotHaveIdenticalImplementations : MethodsShould
             && firstMethod.Body.IsEquivalentTo(secondMethod.Body, false);
 
         protected override SyntaxToken GetMethodIdentifier(IMethodDeclaration method) => method.Identifier;
-        protected override bool IsExcludedFromBeingExamined(ISymbol nodeContainingSymbol) =>
-            nodeContainingSymbol.Kind != SymbolKind.NamedType && !(nodeContainingSymbol is INamespaceSymbol namespaceSymbol
-                                                                   && namespaceSymbol.IsGlobalNamespace
-                                                                   && namespaceSymbol.GetMembers()
-                                                                                     .OfType<INamedTypeSymbol>()
-                                                                                     .Any(x => x.IsTopLevelProgram()));
+        protected override bool ShouldBeAnalyzed(ISymbol nodeContainingSymbol) =>
+            nodeContainingSymbol.Kind == SymbolKind.NamedType
+            && nodeContainingSymbol is INamespaceSymbol namespaceSymbol
```

Comment:
Still not understandable.  Why in this case this should be INamespaceSymbol and not IMethod (that we used before)? Can't we follow the same pattern and allow only in case it is IMethodSymbol (only to be able to process top level file case)? 

Anyways, I think that this part of the code deserves some comments.